int32 kuuve_flag
